MagazineLayout/Public/MagazineLayout.swift

```diff
@@ -688,36 +674,59 @@ public final class MagazineLayout: UICollectionViewLayout {
-
-      let targetYOffsetBeforeUpdate: CGFloat?
-      if let targetContentOffsetAnchor {
-        targetYOffsetBeforeUpdate = yOffset(for: targetContentOffsetAnchor)
-      } else {
-        targetYOffsetBeforeUpdate = nil
-      }
+      let targetContentOffsetAnchor = (
+        layoutStateBeforeRecreateSectionModels ??
+          layoutStateBeforeCollectionViewUpdates ??
+          layoutStateBeforeAnimatedBoundsChange ??
+          self.layoutState
+      ).targetContentOffsetAnchor
+      let targetYOffsetBefore = layoutState.yOffset(for: targetContentOffsetAnchor)
 
       modelState.updateItemHeight(
         toPreferredHeight: preferredAttributes.size.height,
         forItemAt: preferredAttributes.indexPath)
 
-      if let targetYOffsetBeforeUpdate, let targetContentOffsetAnchor {
-        let targetYOffsetAfterUpdate = yOffset(for: targetContentOffsetAnchor)
-        contentOffsetAdjustment = CGPoint(x: 0, y: targetYOffsetAfterUpdate - targetYOffsetBeforeUpdate)
-      } else {
-        contentOffsetAdjustment = nil
+      switch targetContentOffsetAnchor {
+      case .top:
+        context.contentOffsetAdjustment.y = layoutState.minContentOffset.y - layoutState.bounds.minY
+
+      case .bottom:
+        context.contentOffsetAdjustment.y = layoutState.maxContentOffset.y - layoutState.bounds.minY
+
+      case .topItem, .bottomItem:
+        let targetYOffsetAfter = layoutState.yOffset(for: targetContentOffsetAnchor)
+        context.contentOffsetAdjustment.y = targetYOffsetAfter - targetYOffsetBefore
       }
 
       if let attributes = itemLayoutAttributesForPendingAnimations[preferredAttributes.indexPath] {
-        attributes.frame = modelState.frameForItem(
-          at: ElementLocation(indexPath: preferredAttributes.indexPath))
+        switch verticalLayoutDirection {
+        case .topToBottom:
+          attributes.frame = modelState.frameForItem(at: ElementLocation(indexPath: preferredAttributes.indexPath))
+
+        case .bottomToTop:
+          if case .bottom = targetContentOffsetAnchor {
+            attributes.transform = .identity
+            attributes.frame = modelState.frameForItem(at: ElementLocation(indexPath: preferredAttributes.indexPath))
+          } else {
+            let previousHeight = attributes.frame.height
+            attributes.frame = modelState.frameForItem(at: ElementLocation(indexPath: preferredAttributes.indexPath))
+
+            var targetContentOffsetCompensatingYOffsetForAppearingItem = targetContentOffsetCompensatingYOffsetForAppearingItem ?? 0
+            targetContentOffsetCompensatingYOffsetForAppearingItem -= (attributes.frame.height - previousHeight)
+            self.targetContentOffsetCompensatingYOffsetForAppearingItem = targetContentOffsetCompensatingYOffsetForAppearingItem
+            attributes.transform = CGAffineTransform(translationX: 0, y: targetContentOffsetCompensatingYOffsetForAppearingItem)
+          }
+        }
```

Comment:
kind of annoying code that only serves to improve the cell insertion animation when inserting multiple items at the top. I'm still exploring ways to simplify this, but for now, it works well.